proto/site/v1alpha/site.proto

```diff
@@ -0,0 +1,207 @@
+syntax = "proto3";
+
+package com.mintter.site.v1alpha;
+
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/empty.proto";
+
+option go_package = "mintter/backend/genproto/site/v1alpha;site";
+
+// Site API service allows to configure a remote Mintter Site.
+// The server exposing this API should take care about authenticating the caller.
+service Site {
+  // Creates a new invite token for registering a new member.
+  rpc CreateInviteToken(CreateInviteTokenRequest) returns (InviteToken);
+
+  // Redeem a previously created invite token to register a new member.
+  rpc RedeemInviteToken(RedeemInviteTokenRequest) returns (RedeemInviteTokenResponse);
+
+  // Gets public-facing site information.
+  rpc GetSiteInfo(GetSiteInfoRequest) returns (SiteInfo);
+
+  // Updates public-facing site information. Doesn't support partial updates,
+  // hence all the fields must be provided.
+  rpc UpdateSiteInfo(UpdateSiteInfoRequest) returns (SiteInfo);
+
+  // Lists registered members on the site.
+  rpc ListMembers(ListMembersRequest) returns (ListMembersResponse);
+
+  // Gets information about a specific member.
+  rpc GetMember(GetMemberRequest) returns (Member);
+
+  // Deletes an existing member.
+  rpc DeleteMember(DeleteMemberRequest) returns (google.protobuf.Empty);
+
+  // Blocks a given Mintter Account.
+  rpc BlockAccount(BlockAccountRequest) returns (BlockAccountResponse);
+
+  // Unblock a previously blocked Mintter Account.
+  rpc UnblockAccount(UnblockAccountRequest) returns (UnblockAccountResponse);
+
+  // Lists currently blocked Mintter Accounts.
+  rpc ListBlockedAccounts(ListBlockedAccountsRequest) returns (ListBlockedAccountsResponse);
+}
+
+// Request to create a new invite token.
+message CreateInviteTokenRequest {
+  // Required. The role to be associated with the new member after invite is redeemed.
+  // The server might be configured to limit invites for certain roles, but it's not
+  // restricted in this API definition.
+  Member.Role role = 1;
+
+  // Optional. The timestamp after which the invite token will expire if not redeemed.
+  // If not provided, the server will decide the deadline based on the internal configuration.
+  google.protobuf.Timestamp expire_time = 2;
+}
+
+// Request to redeem an invite token. This is the most security-sensitive request,
+// because it allows to register new members on the site. The server must obtain
+// and verify the relation of the communicating peer with a corresponding Mintter Account.
+// After the token is redeemed, a new member on the site must be created, associating
+// the Mintter Account of the caller with the role invite token was created for.
+message RedeemInviteTokenRequest {
+  // Value of the invite token.
+  string token = 1;
```

Comment:
Is the site host included in this token? where does the site host verification happens?
I remember we mentioned this "token" to be a real URL, so anyone can open it on any browser and the site can render a static Tutorial on what to do with this. If this is true, I vote for call this the "Invite URL"

also @ericvicenti mentioned that if is a URL, maybe we need to send it to the backend separate? (host and invite token)

Reply:
oh ok, I get that all this requests go to the remote server? so the host is already included right?